diff --git a/src/common/appbase.cpp b/src/common/appbase.cpp
index 68113d9..79bd6c6 100644
--- a/src/common/appbase.cpp
+++ b/src/common/appbase.cpp
@@ -974,7 +974,7 @@ wxString wxAppTraitsBase::GetAssertStackTrace()
     // don't show more than maxLines or we could get a dialog too tall to be
     // shown on screen: 20 should be ok everywhere as even with 15 pixel high
     // characters it is still only 300 pixels...
-    static const int maxLines = 20;
+    static const int maxLines = 12;
 
     StackDump dump;
     dump.Walk(8, maxLines); // 8 is chosen to hide all OnAssert() calls
