diff --git a/src/tools/darwin.jam b/src/tools/darwin.jam
index edd6b7a..6fa8db0 100644
--- a/src/tools/darwin.jam
+++ b/src/tools/darwin.jam
@@ -25,10 +25,10 @@ import errors ;
 feature framework : : free ;
 
 ## The MacOSX version to compile for, which maps to the SDK to use (sysroot).
-feature macosx-version : : propagated link-incompatible symmetric optional ;
+feature macosx-version : 10.7 : propagated link-incompatible symmetric optional ;
 
 ## The minimal MacOSX version to target.
-feature macosx-version-min : : propagated optional ;
+feature macosx-version-min : 10.7 : propagated optional ;
 
 ## A dependency, that is forced to be included in the link.
 feature force-load : : free dependency incidental ;
